src/ripple/app/ledger/Ledger.cpp

```diff
@@ -334,7 +334,7 @@ bool Ledger::addSLE (SLE const& sle)
 bool Ledger::addTransaction (uint256 const& txID, const Serializer& txn)
 {
     // low-level - just add to table
-    auto item = std::make_shared<SHAMapItem> (txID, txn.peekData ());
+    auto item = std::make_shared<SHAMapItem const> (txID, txn.peekData ());
```

Comment:
I'm also unsure.  It might be safe.  But we've had bugs in this territory in the past.  For this P/R I decided to just follow the existing semantics (make a local shared_ptr).  However I would like to revisit this question in the future in a P/R that more comprehensively follows the lifetime issues of these shared_ptrs.